src/gui/helpers/combat_helper.rs

```diff
@@ -190,60 +190,18 @@ impl GuiHelper for CombatHelper {
-        //     ui.label(format!("HP: {}/{}", enemy.current_hp, enemy.max_hp));
-        //     ui.label(format!("Speed: {}", enemy.speed));
-        //     ui.label(format!("Physical Attack: {}", enemy.physical_attack));
-        //     ui.label(format!("Physical Defense: {}", enemy.physical_defense));
-        //     ui.label(format!("Magical Attack: {}", enemy.magical_attack));
-        //     ui.label(format!("Magical Defense: {}", enemy.magical_defense));
-        //     ui.label(format!("Next Action: {}", enemy.turns_to_action));
-        //     ui.label(format!("Locks: {}", enemy.total_spell_locks));
-        //     for lock in enemy.spell_locks.items.iter() {
-        //         ui.label(format!("{:?}", lock));
-        //     }
-        // }
-
-        // TODO(eein): Another alternative where they are collapsable headers
-        // egui::Grid::new("enemies")
-        //     .min_col_width(10.0)
-        //     .striped(true)
-        //     .show(ui, |ui| {
-
-        //         for (i, enemy) in combat_manager.enemies.items.iter().enumerate() {
-        //            let mut header = format!("NYI ({})  G: {:.5}  U: {:.5}  HP: {}/{}  Act: {}  Locks ({})  ",
-        //                 i,
-        //                 enemy.guid,
-        //                 enemy.unique_id,
-        //                 enemy.current_hp,
-        //                 enemy.max_hp,
-        //                 enemy.turns_to_action,
-        //                 enemy.total_spell_locks);
-
-        //             for lock in enemy.spell_locks.items.iter() {
-        //                 header = format!("{} {:?}", header, lock);
-        //             }
-
-        //             egui::CollapsingHeader::new(header)
-        //                 .default_open(true)
-        //                 .show(ui, |ui| {
-        //                 ui.label(format!("Spd: {}, Patk: {}, Pdef: {}, Matk: {}, Mdef: {}",
-        //                     enemy.speed,
-        //                     enemy.physical_attack,
-        //                     enemy.physical_defense,
-        //                     enemy.magical_attack,
-        //                     enemy.magical_defense,
-        //                 ));
-        //             });
-        //             ui.end_row();
-        //         }
-        //     });
-        // ui.separator();
     }
 }
+
+fn damage_type_image(ui: &mut egui::Ui, damage_type: &CombatDamageType) {
+    let damage_type_icon = match damage_type {
+        CombatDamageType::Arcane => ASSETS.damage_types.arcane.clone(),
+        CombatDamageType::Blunt => ASSETS.damage_types.blunt.clone(),
+        CombatDamageType::Moon => ASSETS.damage_types.moon.clone(),
+        CombatDamageType::Poison => ASSETS.damage_types.poison.clone(),
+        CombatDamageType::Sun => ASSETS.damage_types.sun.clone(),
+        CombatDamageType::Sword => ASSETS.damage_types.sword.clone(),
+        _ => ASSETS.stats.hp.clone(),
+    };
+    ui.add(damage_type_icon);
```

Comment:
cloning the assets here only clones the handle or? This would presumably be slow if it's copying the entire image asset each redraw.

Reply:
got yelled at by clippy - i think its cloining the texture its going to draw - i'll see if i can fix it